### PR TITLE
Add `on_auth_fail` callback

### PR DESCRIPTION
**Pull request by red-001**
Called when a client fails to supply the correct password for the account it's attempting to login as.

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -2615,6 +2615,10 @@ Call these functions only at load time!
 * `minetest.register_on_leaveplayer(func(ObjectRef, timed_out))`
     * Called when a player leaves the game
     * `timed_out`: True for timeout, false for other reasons.
+* `minetest.register_on_auth_fail(func(ip, name))`
```
Comment (SmallJoker) [inline doc/lua_api.txt]: Please use `name, ip` as in `minetest.register_on_prejoinplayer(func(name, ip))`

**Comment by SmallJoker** [comment]
What are the use cases of this callback? Network delays make it quite hard to crack a password of users, if that's what you would like to prevent.

**Comment by paramat** [comment]
Yes please i'd like to know why this is needed.

**Comment by sofar** [comment]
Possible use: Telling uses that someone from another IP address was attempting to log in the account, which helps in maintaining legitimate bans for IP addresses.

**Comment by red-001** [comment]
Like sofar said it allows informing users that someone was trying to access their account and while cracking even a mediocre password should be next to impossible over the network, a lot of users do use weak passwords. Plus if someone did just forget their password it allows informing them of any password reset system the server might have.

**Comment by sofar** [comment]
I am the author of `tallow` which is something similar to `fail2ban`. Essentially, having this callback could enable the very same thing, and solutions like this are reasonable. Without this call, people have to resort to much more complex methods.

File: doc/lua_api.txt (hunk 1 above)
Comment (red-001) [reply doc/lua_api.txt]: done

**Pull request merged**